xbmc/ServiceManager.cpp

```diff
@@ -229,6 +232,11 @@ bool CServiceManager::StartAudioEngine()
 // stage 3 is called after successful initialization of WindowManager
 bool CServiceManager::InitStageThree()
 {
+  m_stereoscopicsManager.reset(new CStereoscopicsManager(*m_settings,
+                                                         *m_dataCacheCore,
+                                                         GetRenderSystem()));
```

Comment:
tis is wrong. rendersystem can change

Reply:
please explain

Reply:
as mentioned, render system can change. having a WinSystem is optional. Kodi can (will) support headless, X11, Wayland by a single binary.
further all services are subject to change. this is why ServiceBroker will return pointers

Reply:
> further all services are subject to change.

Do you mean here that any service can be replaced or removed at any time?

> this is why ServiceBroker will return pointers

For the "being optional" part, I don't see why clients shouldn't get references for the things they require and pointers for the optional stuff.

But maybe there is no point in discussing this further if #13234 will change everything eventually anyway?

Reply:
> Do you mean here that any service can be replaced or removed at any time?

yes. services should be losely coupled, taking into consideration that the other ends may not be available. you see the mess we have with creating services. this is uncontrollable in the current state.

> But maybe there is no point in discussing this further if #13234 will change everything eventually anyway?

correct